md/bdm-in-rest-api.md

```diff
@@ -216,6 +216,108 @@ class CarManagement implements RestApiController {
     }
 ```
 
+### Returning the whole object with an API link load in the lazy fields
+
+The idea is to create a custom Json serializer.  
+A custom Json serializer is a class which extends *com.fasterxml.jackson.databind.JsonSerializer*. There is a method *serialize* to implement, which has the responsability to serialize the input model into Json.  
+The custom Json serializer has to come with an other classe, an object mapper,  which extends *com.fasterxml.jackson.databind.ObjectMapper*.  
+This mapper register a simple module (*com.fasterxml.jackson.databind.module.SimpleModule*), which has to contains the custom serializer.  
+At the end, in your rest API endpoint, you interact with the mapper.
+
+Here is an implementation example for the object Car which has four lazy attributes of type Wheel:  
+
+The serializer takes a list of Car in input, and build a Json object for each car. The wheels are replaced with links to an other Rest API extension with the car ID and the wheel number in parameter. Calling this API will return the wheel. This is a classic lazy behavior.
+```groovy
+/***********************
+ ***** SERIALIZER ******
+ ***********************/
+
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.databind.JsonSerializer
+import com.fasterxml.jackson.databind.SerializerProvider
+
+class CarSerializer extends JsonSerializer<List<Car>>{
+
+	@Override
+	public void serialize(List<Car> cars, JsonGenerator jgen, SerializerProvider provider)throws IOException, JsonProcessingException {
+		jgen.writeStartArray()
+		
+		cars.forEach({serializeCar(it,jgen)})
+		
+		jgen.writeEndArray()
+		
+	}
+	
+	private void serializeCar(Car car, JsonGenerator jgen) {
+		jgen.writeStartObject()
+		
+		jgen.writeNumberField("carID", car.getPersistenceId())
+		jgen.writeStringField("model", car.getModel())
+		jgen.writeNumberField("buildYear", car.getBuildYear())
+		jgen.writeStringField("color", car.getColor())
+		jgen.writeStringField("wheel1Request", getWheelRequest(car.getPersistenceId(),1))
```

Comment:
why not introduce a carPersistenceId variable at it is used 4 times?

Reply:
also, put an empty space after the last comma to match our formatting rules

Reply:
i never declare a variable for small getters, it doesn't improve perf neither readability (in my view)